src/nemdeling/nemdeling.service.ts

```diff
@@ -0,0 +1,153 @@
+import { Injectable, Logger } from "@nestjs/common";
+import { DisplayApiService } from "../display-api/display-api.service";
+import { PlaylistPlaylistJsonld, PutV1PlaylistSlideIdRequestInner } from "../display-api-client";
+import { NemdelingSlide } from "./types";
+import { PlaylistSlide } from "../display-api/types";
+
+@Injectable()
+export class NemDelingService {
+  constructor(
+    private readonly displayApiService: DisplayApiService,
+    private readonly logger: Logger
+  ) {}
+
+  /**
+   * If service messages are syncing.
+   *
+   * 0 means that no sync is in progress.
+   * A value greater than 0 means that syncing is in progress, the number is the
+   * number og requests that has been skipped.
+   */
+  public serviceMessagesAreSyncing = 0;
+
+  /**
+   * Gets a service message playlist from a given screen name.
+   *
+   * @param {string} screenName the name of the screen to get the playlist for.
+   */
+  async getServiceMessagePlaylistFromScreenName(
+    screenName: string
+  ): Promise<PlaylistPlaylistJsonld | null> {
+    const playlistName = `service_message_${screenName}`;
+    return await this.displayApiService.getPlaylistByName(playlistName);
+  }
+
+  /**
+   * Create slide if it doesn't already exist
+   *
+   * @param {NemdelingSlide} slide - The slide as it should exist.
+   * @param {number} weight - The weight the slide should have on the playlist.
+   * @param {PlaylistSlide[]} playlistSlides - The list of existing slides on playlist.
+   */
+  ensureSlideOnPlaylist(
+    slide: NemdelingSlide,
+    weight: number,
+    playlistSlides: PlaylistSlide[]
+  ): Promise<PutV1PlaylistSlideIdRequestInner> {
+    return new Promise((resolve, reject) => {
```

Comment:
Any reason not to do this in a async method and await?

Reply:
This allow us to make the requests in parallel.